### ISSUE TITLE
Exception raised: Psych::DisallowedClass

### ISSUE THREAD

**Issue by btihen** [bridgetownrb/bridgetown#593]
Ruby library:  Psych seems to need to also allow Symbol - not sure why, but where would I add this to the list of permitted 

```
❯ bin/bridgetown start                                              
[Bridgetown]           Starting: Bridgetown v1.0.0 (codename "Pearl")
[Bridgetown]   Exception raised: Psych::DisallowedClass
[Bridgetown] Tried to load unspecified class: Symbol 
[Bridgetown]                  1: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:99:in `find'
[Bridgetown]                  2: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:28:in `load'
[Bridgetown]                  3: (eval):2:in `symbol'
[Bridgetown]                  4: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:32:in `symbolize'
[Bridgetown]                  5: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:84:in `symbolize'
[Bridgetown]          Backtrace: Use the --trace option for complete information.
```

```
❯ bin/bridgetown start
[Bridgetown]           Starting: Bridgetown v1.1.0.beta3 (codename "Belmont")
[Bridgetown]   Exception raised: Psych::DisallowedClass
[Bridgetown] Tried to load unspecified class: Symbol 
[Bridgetown]                  1: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:99:in `find'
[Bridgetown]                  2: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:28:in `load'
[Bridgetown]                  3: (eval):2:in `symbol'
[Bridgetown]                  4: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:32:in `symbolize'
[Bridgetown]                  5: /Users/wti0405/.rbenv/versions/3.1.2/lib/ruby/3.1.0/psych/class_loader.rb:84:in `symbolize'
[Bridgetown]          Backtrace: Use the --trace option for complete information.
```

**Bridgetown Version**:

Bridgetown v1.0.0 (codename "Pearl")
& 
Bridgetown v1.1.0.beta3 (codename "Belmont")

**Current behavior**
The problem is in `bridgetown.config.yml` with:
```
collections:
  posts:
   permalink: :categories/:name/
```

It is fixable by starting the permalink with a '/':
```
collections:
  posts:
   permalink: /:categories/:name/
```

**Expected behavior**
Successfully start or a clear error message what is misconfigured - ideally the file and line.

I found the error with a git diff and noticed the suspect line in a yaml file. A clearer error message on load failure would be helpful

**Computing environment (please complete the following information):**
 - OS: MacOS 12.4 (M1)
 - Ruby Version:
   - ruby 3.0.4p208 (2022-04-12 revision 3fa771dded) [arm64-darwin21]
   - ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]

**Additional context**

**Comment by jaredcwhite** [comment]
Yeah there's been some weird stuff going on with loading YAML lately in the Ruby ecosystem. Not fun. I'll take a look at this ASAP

**Comment by btihen** [comment]
Thanks - I found the error, and can start now, but I think if possible listing the file and line number would help.

I found by reviewing the diff and found a line that could be seen as a symbol in a yaml file. 
Given that Bridgetown uses ':' in yaml - maybe permalinks should by default require a start with '/' or detect staring with ':' would be invalid - not sure how easy working with Psych and Error capturing is.

**Comment by jaredcwhite** [comment]
> Thanks - I found the error, and can start now, but I think if possible listing the file and line number would help.
> 
> I found by reviewing the diff and found a line that could be seen as a symbol in a yaml file. Given that Bridgetown uses ':' in yaml - maybe permalinks should by default require a start with '/' or detect staring with ':' would be invalid - not sure how easy working with Psych and Error capturing is.
> 
> Anyway, a bit confusing. If you can point me to where these files are validated or load errors are captured (or suggestions how to change), I'd be glad to look at creating a PR

Ah yeah, anything starting with a `:` should be in quotes if it's supposed to be a string:

```yaml
some:
  value: ":this is a string"
```

Was that something you found in Bridgetown's codebase directly, or in your own project?

**Comment by btihen** [comment]
all works with a `bridgetown new` - I just made a permalink without the leading '/' - which wan't needed in links NOT STARTING with ':' - so definately my error in the config, however, I think the error message could be more helpful in finding this without first a `bridgetown new` and `git diff`

**Comment by btihen** [comment]
I just noticed a similar problem with the MD parsing.  When there is an error - it reports the file, and crashes.  The line or the text would be helpful instead of removing text until the problem goes away and then adding text back until it returns - just not as effective as git bisect.  

Error handling with the file, context and ideally line number would be very helpful.

**Comment by jaredcwhite** [comment]
I've added a bit better error messaging for when there's a symbol value in `bridgetown.config.yml`. I'd happily show the line number, but unfortunately Psych doesn't seem to provide that information.

Closed in 99f1215092168b0414af189fb1763bbec955b5d6
